libr/core/disasm.c

```diff
@@ -4379,15 +4379,22 @@ static void ds_print_esil_anal(RDisasmState *ds) {
 				// function name not resolved
 				nargs = DEFAULT_NARGS;
 				if (fcn) {
-					nargs = fcn->nargs;
+					// @TODO: fcn->nargs should be updated somewhere and used here instead
+					nargs = r_anal_var_count (core->anal, fcn, 's', 1) +
+							r_anal_var_count (core->anal, fcn, 'b', 1) +
+							r_anal_var_count (core->anal, fcn, 'r', 1);
 				}
 				if (nargs > 0) {
-					ds_comment_middle (ds, "; CALL: ");
+					if (fcn_name) {
+						ds_comment_middle (ds, "; %s(", fcn_name);
+					} else {
+						ds_comment_middle (ds, "; 0x%"PFMT64x"(", pcv);
+					}
 					for (i = 0; i < nargs; i++) {
-						ut64 v = r_debug_arg_get (core->dbg, R_ANAL_CC_TYPE_STDCALL, i);
+						ut64 v = r_debug_arg_get (core->dbg, R_ANAL_CC_TYPE_FASTCALL, i);
```

Comment:
we should take the type of calling conventions from the function information, instead of harcoding std or fast here

Reply:
Do you mean the stack emulation is not working in there? can be possible.., so we can look at this later. fastcall is pretty common and we can improve that later with tests.